low-level/sensors/impl/generic/node_hw.py

```diff
@@ -554,6 +607,8 @@ def _notify_NodeDataMsgHandler(self):
 
         if last_index is not None:
             self._write_index_file(last_index)
+        else:
```

Comment:
Doesnt need to be under 'else' case

After SEL processing is done, check for faulty resources explicitly. What was faulty resource last fault recorded, and get sensor/sdr directly and see if its rectified, if yes, raise required alert & remove from faulty alerts list

Reply:
Done